#!/usr/bin/env python
# -*-coding:utf-8-*-
PACKAGE = "laser_detector"

from dynamic_reconfigure.parameter_generator_catkin import * 
gen = ParameterGenerator()

gen.add("is_debug", bool_t, 0, "Is Debug", True)

gen.add("laser_remove_front", double_t, 0, "Laser Remove Front", 0.05, -999, 999)
gen.add("laser_remove_back", double_t, 0, "Laser Remove Back", -0.75, -999, 999)
gen.add("laser_remove_left", double_t, 0, "Laser Remove Left", 0.35, -999, 999)
gen.add("laser_remove_right", double_t, 0, "Laser Remove Right", -0.35, -999, 999)

gen.add("laser_detect_front", double_t, 0, "Laser Detect Front", 2.0, -999, 999)
gen.add("laser_detect_back", double_t, 0, "Laser Detect Back", -2.5, -999, 999)
gen.add("laser_detect_left", double_t, 0, "Laser Detect Left", 2.0, -999, 999)
gen.add("laser_detect_right", double_t, 0, "Laser Detect Right", -2.0,-999, 999)
gen.add("laser_detect_top", double_t, 0, "Laser Detect Top", 2.0, -999, 999)
gen.add("laser_detect_bottom", double_t, 0, "Laser Detect Bottom", -1.5, -999, 999)

gen.add("enable_radius_outlier_filter", bool_t, 0, "Enable Radius Outlier", True)
gen.add("radius_outlier_filter_nums", double_t, 0, "Radius Outlier Filter Nums", 10, 0, 999)
gen.add("radius_outlier_filter_radius", double_t, 0, "Radius Outlier Filter Radius", 0.2, 0, 999)

exit(gen.generate(PACKAGE, "laser_detector", "LaserDetector"))
